include/pgduckdb/types/pgduckdb_enum.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include "pgduckdb/duckdb_vendor/enum_type_info_templated.hpp"
+
+namespace pgduckdb {
+
+using duckdb::EnumTypeInfo;
+using duckdb::EnumTypeInfoTemplated;
+using duckdb::ExtraTypeInfo;
+using duckdb::InternalException;
+using duckdb::LogicalType;
+using duckdb::LogicalTypeId;
+using duckdb::make_shared_ptr;
+using duckdb::PhysicalType;
+using duckdb::shared_ptr;
+using duckdb::Vector;
+
+template <class T>
+class PGDuckDBEnumTypeInfo : public EnumTypeInfoTemplated<T> {
+public:
+	PGDuckDBEnumTypeInfo(Vector &values_insert_order_p, idx_t dict_size_p, Vector &oid_vec)
+	    : EnumTypeInfoTemplated<T>(values_insert_order_p, dict_size_p), oid_vec(oid_vec) {
+	}
+
+public:
+	const Vector &
+	GetMemberOids() const {
+		return oid_vec;
+	}
+
+private:
+	Vector oid_vec;
+};
+
+struct PGDuckDBEnum {
```

Comment:
Sure I'll add the comments 👍 

I can't extend `EnumTypeInfo` without subclassing, which I'm doing with `PGDuckDBEnumTypeInfo`, but the internals of `EnumTypeInfo` are assuming (in things like `EnumTypeInfo::GetPos`) that the derived class is `EnumTypeInfoTemplated` and it will cast to that.

Since `PGDuckDBEnumTypeInfo` is a templated class I can't add a static method there[1]
So the next best option is to add `PGDuckDBEnum` just to contain static methods (probably should move the enum-related methods from `pgduckdb_types.cpp` into this as well for cohesion)


[1] I can do this, but then it has to be called like `PGDuckDBEnumTypeInfo<uint8_t>::CreateType(...)` even if we don't know what type we're creating yet.
Which either looks confusing or we would need to move that if/else tree outside of the function, which defeats the purpose of having this static function.